crates/context_aware_config/migrations/2024-04-22-122806_config_verions/up.sql

```diff
@@ -0,0 +1,14 @@
+-- Your SQL goes here
+-- Name: functions; Type: TABLE; Schema: public; Owner: -
+--
+CREATE TABLE public.config_versions (
+    id bigint PRIMARY KEY,
+    config json NOT NULL,
+    config_hash TEXT NOT NULL,
+    tags varchar(100)[] check (array_position(tags, null) is null),
+    created_at timestamp without time zone DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS config_verions_tags_index ON public.config_versions USING gin(tags);
```

Comment:
yes we can write db query , at the end it will generate sql statement only, index is added for fast retrieval since there can be multiple rows and with array of tags , performance wise it will be better

Reply:
No the places where this index benefits us is not supported 
https://www.postgresql.org/docs/current/arrays.html
it impacts us on the writes though. We should remove this indexing